subsys/ipc/ipc_service/backends/Kconfig

```diff
@@ -44,5 +43,30 @@ config IPC_SERVICE_STATIC_VRINGS_ALIGNMENT
 	help
 	  Static VRINGs alignment
 
-rsource "Kconfig.icmsg"
+config IPC_SERVICE_ICMSG
+	bool
+	select SPSC_PBUF
+	select SPSC_PBUF_USE_CACHE
+	help
+	  Icmsg library
+
+if IPC_SERVICE_ICMSG
+
+config ICMSG_CB_BUF_SIZE
+	int "Size of callback buffer size"
+	range 1 65535
+	default 255
+	help
+	  Size of callback buffer used for processing received data in work
+	  queue thread. If you are sure that your application never sends data
+	  data bigger than some size, you can safely change this option to
+	  reduce RAM consumption in your application.
+
+# The Icmsg library in its simplicity requires the system workqueue to execute
+# at a cooperative priority.
+config SYSTEM_WORKQUEUE_PRIORITY
+	range -256 -1
+
+endif # IPC_SERVICE_ICMSG
```

Comment:
Can we make this `depends on IPC_SERVICE_ICMSG` instead? Also I wonder whether at this point we need something like `Kconfig.library` to host all these configs.

Also I think this should be `IPC_SERVICE_ICMSG_..` for consistency.

Reply:
I'll rename to `IPC_SERVICE_ICMSG_CB_BUF_SIZE` for consistency, but moving configs to separate `Kconfig.library` I would like to perform in a separate PR to speed up this one. If that's ok for you

Reply:
sure, I'm just bike-shedding here

Reply:
done, alongside replacing `if` with `depends on`